tgui/packages/tgui/interfaces/Biogenerator.js

```diff
@@ -0,0 +1,247 @@
+import { useBackend, useSharedState } from "../backend";
+import {
+  Button,
+  Section,
+  Box,
+  Flex,
+  Icon,
+  Collapsible,
+  NumberInput,
+  ProgressBar,
+  Dimmer} from "../components";
+import { Window } from "../layouts";
+
+export const Biogenerator = (props, context) => {
+  const { data } = useBackend(context);
+  const { container } = data;
+
+  return (
+    <Window>
+      <Window.Content display="flex" className="Layout__content--flexColumn">
+        {!container ? (
+          <MissingContainer />
+        ) : (
+          <>
+          <Processing />
+          <Storage />
+          <Controls />
+          <Products />
+          </>
+        )}
+      </Window.Content>
+    </Window>
+  );
+};
+
+const MissingContainer = (props, context) => {
+  return (
+    <Section flexGrow="1">
+      <Flex height="100%">
+        <Flex.Item
+          grow="1"
+          textAlign="center"
+          align="center"
+          color="silver">
+          <Icon
+            name="flask"
+            size="5"
```

Comment:
Forgot to clean this sort of thing up. There's two styles: `={5}` and also `="5px"` or whatever. I want to standardize on using `px`.

Reply:
yeah, we should really make a repo standard for how we want to format this stuff. I just suggested this because this is the standard you're using for icons in the rest of the file, it also doesn't have a unit attached to it so might aswell.